src/domain/alerts/alerts.repository.ts

```diff
@@ -117,18 +132,53 @@ export class AlertsRepository implements IAlertsRepository {
     }
   }
 
-  private async _notifyUnknownTransaction(
-    chainId: string,
-    log: AlertLog,
-  ): Promise<void> {
-    const emails = await this.emailRepository.getVerifiedEmailsBySafeAddress({
-      chainId,
-      safeAddress: log.address,
+  async getRecoveryTarget(args: {
```

Comment:
Overall I think this is the way to go, but I'd like to have some separation of layers here. I'd consider the blockchain as another data source for us, and I'd extract this functionality fo the `datasources` layer, by creating something like a `BlockchainAPI/OnChainAPI` (the naming can be improved 🙂) with its interface on the domain side, instead of doing this data retrieval here right away. Wdyt? (cc. @fmrsabino)

We could also leave that for a followup PR I guess.

Reply:
I agree that a separate data source makes the most sense. As this [area of the code is evolving](https://github.com/safe-global/safe-client-gateway/pull/922), I will look into creating a new data source that we could then use here.

We can keep this open for discussion until then.